### PR TITLE
DRILL-1491: Support for JDK 8

### PR DESCRIPTION
**Pull request by vladimirtkach**
Changed jdk version from 7 to 8 in pom.xml, drill-config.sh and others

### CODE REVIEW

**Comment by ilooner** [comment]
@vladimirtkach This test is failing on Travis:

```
  TestOutputBatchSize.testFlattenUpperLimit:695->getExpectedSize:58->PhysicalOpUnitTestBase.getReaderListForJsonBatches:483->PhysicalOpUnitTestBase.getRecordReadersForJsonBatches:479->PhysicalOpUnitTestBase.getJsonReadersFromBatchString:505 Â» TestTimedOut
```

File: .travis.yml [hunk 1]
```diff
@@ -17,9 +17,9 @@ before_install: git fetch --unshallow
...
 language: java
 jdk:
-  - openjdk7
+  - openjdk8
 cache:
   directories:
   - "$HOME/.m2"
 install: MAVEN_OPTS="-Xms1G -Xmx1G" mvn install --batch-mode -DskipTests=true -Dmaven.javadoc.skip=true -Dmaven.source.skip=true
-script: mvn install -DexcludedGroups="org.apache.drill.categories.SlowTest,org.apache.drill.categories.UnlikelyTest,org.apache.drill.categories.SecurityTest" -DforkCount=1 -DmemoryMb=2560 -DdirectMemoryMb=4608
+script: mvn install -DexcludedGroups="org.apache.drill.categories.SlowTest,org.apache.drill.categories.UnlikelyTest,org.apache.drill.categories.SecurityTest" -DforkCount=1 -DmemoryMb=2560 -DdirectMemoryMb=6008
```
Comment (ilooner) [inline .travis.yml]: @vladimirtkach Why the change in direct memory for travis? Did you run into an issue with the previous setting?

Comment (vladimirtkach) [reply .travis.yml]: @ilooner yes , there was:
testFlattenUpperLimit(org.apache.drill.exec.physical.unit.TestOutputBatchSize)  Time elapsed: 260.527 sec  <<< ERROR!
java.lang.OutOfMemoryError: GC overhead limit exceeded

For now changed to previous value.

File: exec/java-exec/src/test/java/org/apache/drill/exec/server/TestDrillbitResilience.java [hunk 2]
```diff
@@ -650,6 +650,7 @@ public void dataArrived(final QueryDataBatch result, final ConnectionThrottle th
 
   @Test // DRILL-2383: Cancellation TC 3: cancel after all result set are produced but not all are fetched
   @Repeat(count = NUM_RUNS)
+  @Ignore
```
Comment (parthchandra) [inline exec/java-exec/src/test/java/org/apache/drill/exec/server/TestDrillbitResilience.java]: Please log a JIRA and reference it in this ignore statement

Comment (vladimirtkach) [reply exec/java-exec/src/test/java/org/apache/drill/exec/server/TestDrillbitResilience.java]: @parthchandra added jira to ignore annotation https://issues.apache.org/jira/browse/DRILL-6228

**Comment by parthchandra** [comment]
+1
